### PR TITLE
Generalize Menu

### PR DESCRIPTION
**Pull request by matthewrsj**
This uses Jinja2 templating and page metadata to generate a main menu. Exact metadata requirements are listed in this branch's README.

Remaining issues:
- Cannot include externally linked menu items
- Cannot include direct template pages (i.e. osuosl pelican blog page)

**work in progress**

### CODE REVIEW

File: templates/includes/menu.html [hunk 1]
```diff
@@ -47,72 +47,51 @@
...
-                                    <li class="menu-1821 menu-path-node-538 first odd "><a href="/sponsors">Sponsors</a></li>
-                                    <li class="menu-1841 menu-path-node-133  even last"><a href="/donate/why-do-we-give-osuosl">Why Do We Give to the OSL?</a></li>
-                                </ul>
-                            </li>
-                        </ul>
-                    </li>
-                </ul>
-            </div>
-        </div>
+<div id='main-menu' role='navigation'>
```
Comment (lucywyman) [inline templates/includes/menu.html]: Can you add this logic to the mobile menu as well?

**Comment by lucywyman** [comment]
@matthewrsj  I pushed a commit to add the same logic to the mobile menu and fix indentation -- lemme know what you think! I can always pull the commit out of here if need be.

**Comment by matthewrsj** [comment]
Ok, this looks really good. You fixed the footer as well. We should wait to merge this until we figure out the external links and direct templates.

What if we do away with direct templates altogether and just have a restructuredtext file with all the html in a ``.. raw:: html`` directive like I do with the [forms](https://raw.githubusercontent.com/osuosl/osuosl-pelican/master/content/forms/power-request-hosting.rst) pages?

**Comment by lucywyman** [comment]
It's ugly, but I'd be ok with it. I think we only have that one page :)

**Comment by matthewrsj** [comment]
I just found a bug with the logic when introducing the cass-pelican menu. cass-pelican has the following structure:

```
Home   About   Organization <-------------|
               Staff                      |
               Researchers                |
               Students                   | ------------ Organization and Units point to the same place
               Alumni                     |
               Projects                   |
               Units <--------------------|
                   Open Source Lab
                   Software Development Group
                   Test Lab
                   IoT Test and Compatibility Lab
```

This is an issue because ``duplicate_name`` only places the menu item directly below the parent, not at the end of the list and certainly not with a submenu. We need to integrate the following for this to work:

```
duplicate_name
duplicate_order
duplicate_menu_gchild
```
I think

**Comment by matthewrsj** [comment]
@lucywyman as I start working on this, I'm realizing [that solution](https://github.com/osuosl/dougfir-pelican-theme/pull/12#issuecomment-143887235) may be more difficult than it seems since it would require processing the page twice in an undetermined order. 

What if we just *made a duplicate page* with slightly different metadata (but no ``duplicate_*``)? Keep in mind we would need different slugs so the url would be slightly different.

**Comment by matthewrsj** [comment]
Forget everything! Do this instead with Python:
```
{
        {'title': 'A Top Level Menu Item',
         'weight': 0,
         'link': <the page this points to>,
        'children': []},
        {'title': 'Another Top Level Menu Item',
         'weight': 0,
         'link': <the page this points to>,
        'children': [
                {'title': 'Child One',
                 'weight': 0,
                 'link': <the page this points to>,
                 'children': []},
                {'title': 'Child Two',
                 'weight': 15,
                 'link': <the page this points to>,
                 'children': []},
                ]
        },
        {'title': 'A Top Level Menu Item',
         'weight': 0,
         'link': <the page this points to>,
        'children': []},
        ...
}
```

**Comment by lucywyman** [comment]
@matthewrsj  Alright -- It's not pretty, but it gets the job done. You basically manually add the menu_filter data structure to the pelicanconf, which isn't super scalable but for just a couple of template pages it should hold up.  Then it's just included in the menu list as usual.

**Comment by matthewrsj** [comment]
I'm getting this error:

```
CRITICAL: UndefinedError: 'pelican.contents.Page object' has no attribute 'order'
```

Which is weird since we no longer rely on the 'order' attribute. I will keep investigating, it must be something on my end.

**Comment by matthewrsj** [comment]
Ok, 006ffd5a7e74c2badbc70f18d6732b0953a3b66b allows the direct templates to be added to the menu structure depending on their weight instead of being appended to the end. Works with this version of menufilter: https://github.com/osuosl/osuosl-pelican/commit/a7c34fc069ac0e9d72b55e960c5d031df86533e8

**Comment by lucywyman** [comment]
:+1: Everything looks good on my end!  Glad we finally found a solution.

**Comment by matthewrsj** [comment]
+1 from me too, but since we wrote this pr we should have someone else look at it. @Kennric

**Comment by matthewrsj** [comment]
Still +1 from me, I just updated the README to explain how to include direct templates in the menu structure.

**Comment by ElijahCaine** [comment]
@matthewrsj This looks good except for the lack of mobile menu (tested on Firefox's mobile preview).
I know that's not really in the PRs scope, but it's concerning.
Also there is no content on the front page below the slideshow. I'm not sure what *is* supposed to be there, but its just header, slideshow, side-nav, footer on the page.

... the menus look great though.

**Comment by matthewrsj** [comment]
@ElijahCaine mobile menu is another issue (sort of, we need to test on mobile then address concerns). Also we aren't putting content below the slideshow.

**Comment by lucywyman** [comment]
@Kennric  I think you're the last one we need a +1 from before merging, if you get a chance!

File: templates/base.html [hunk 2]
```diff
@@ -5,10 +5,6 @@
 </head>
 <body class="html front not-logged-in one-sidebar sidebar-first page-node " >
 {% include 'includes/top-hat.html' %}
-<!--  I'm not sure what this is here for
```
Comment (Kennric) [inline templates/base.html]: It's good practice if you have header, slideshow, etc at the top of your page and content below - this allows people using screen readers or other assistive technology to skip all the stuff they can't see anyway and get to the content. It requires a #content anchor where the content begins.

**Comment by Kennric** [comment]
Looks pretty good - at some point we should audit this theme for accessibility and add skip-links, etc, but that doesn't need to happen in this PR. +1

File: templates/base.html (hunk 2 above)
Comment (lucywyman) [inline templates/base.html]: Ohhhhhh, ok.  I can uncomment it then!

**Pull request merged**